api/v1/storagecluster_types.go

```diff
@@ -294,6 +296,8 @@ type ManageCephObjectStores struct {
 	// +kubebuilder:validation:MaxLength=253
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
 	StorageClassName string `json:"storageClassName,omitempty"`
+	// DataPoolSpec specifies the pool specification for the default cephObjectStore data pool
+	DataPoolSpec rookCephv1.PoolSpec `json:"dataPoolSpec,omitempty"`
```

Comment:
How about if we make these nullable? Then it's simple to know if the user specified a spec or not, and we can write to the log to make it more obvious for troubleshooting, in case the user applied unsupported settings.
```suggestion
	DataPoolSpec *rookCephv1.PoolSpec `json:"dataPoolSpec,omitempty"`
```

Reply:
resolving this as oer the discussion on the other thread